tests/networkpolicy_test.go

```diff
@@ -1,173 +1,285 @@
 package tests_test
 
 import (
+	"fmt"
 	"time"
 
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
-	v1network "k8s.io/api/networking/v1"
+	corev1 "k8s.io/api/core/v1"
+	networkv1 "k8s.io/api/networking/v1"
 	"k8s.io/apimachinery/pkg/api/errors"
-	v13 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	netutils "k8s.io/utils/net"
 
 	v1 "kubevirt.io/client-go/api/v1"
 	"kubevirt.io/client-go/kubecli"
 	"kubevirt.io/kubevirt/tests"
 	cd "kubevirt.io/kubevirt/tests/containerdisk"
+	"kubevirt.io/kubevirt/tests/libvmi"
 )
 
 var _ = Describe("[rfe_id:150][crit:high][vendor:cnv-qe@redhat.com][level:component]Networkpolicy", func() {
 
-	var virtClient kubecli.KubevirtClient
+	var (
+		virtClient       kubecli.KubevirtClient
+		vmia, vmib, vmic *v1.VirtualMachineInstance
+
+		curlCheckCmd        = "curl --head %s:%d --connect-timeout 5\n"
+		port80              = intstr.FromInt(80)
+		port81              = intstr.FromInt(81)
+		tcp                 = corev1.ProtocolTCP
+		networkpolicyport80 = networkv1.NetworkPolicyPort{
+			Port:     &port80,
+			Protocol: &tcp,
+		}
+		networkpolicyport81 = networkv1.NetworkPolicyPort{
+			Port:     &port81,
+			Protocol: &tcp,
+		}
```

Comment:
We use these far below. For me as a reviewer this opens more questions than asnwer. Could we just have these in-line in their respective tests? We already have most of the Policy in there.

Reply:
Done